src/index.es.ts

```diff
@@ -11,8 +11,12 @@ import {
 } from './helpers';
 import { createInfiniteHitsSessionStorageCache } from './lib/infiniteHitsCache';
 
-const instantsearch = (options: InstantSearchOptions): InstantSearch =>
-  new InstantSearch(options);
+const instantsearch = <
+  TUiState = Record<string, unknown>,
+  TRouteState = TUiState
+>(
+  options: InstantSearchOptions<UiState & TUiState, TRouteState>
```

Comment:
I was just thinking, should the argument maybe be TIndexUiState (so you can & all custom widgets together). The downside I see is that would mean the argument TRouteState likely still would be a full ui state (key per index)

Reply:
How would we get the full UI state typing if users only provide `IndexUiState`?

Reply:
It would be possible like this: `InstantSearchOptions<UiState & { [indexName: string]: TIndexUiState] }, TRouteState>`

Reply:
And what if you have multiple indices that don't share the same index UI state?

Reply:
Good point, let's keep it as suggested in the PR then! UiState is essentially always optional, but I see why you'd want that